src/OrchardCore/OrchardCore.DisplayManagement/Views/ShapeViewModelOfT.cs

```diff
@@ -3,6 +3,12 @@ namespace OrchardCore.DisplayManagement.Views
     public class ShapeViewModel<T> : ShapeViewModel
     {
         public ShapeViewModel(T value)
+            : this(typeof(T).Name, value)
+        {
+        }
+
+        public ShapeViewModel(string shapeType, T value)
+            : base(shapeType)
```

Comment:
Add param check  should not be null or empty

Reply:
added in the base class